pow/src/main/java/tech/pegasys/artemis/pow/event/Deposit.java

```diff
@@ -78,37 +82,56 @@ public long getAmount() {
     return amount;
   }
 
+  @Override
+  public void filterOutputFields(List<String> outputFields) {
+    this.outputFieldMap.put("eventType", "Deposit");
+    for (String field : outputFields) {
+      switch (field) {
+        case "pubkey":
+          this.outputFieldMap.put(
+              "pubkey", pubkey.getPublicKey().toBytesCompressed().toHexString());
+          break;
+
+        case "withdrawal_credentials":
+          this.outputFieldMap.put("withdrawal_credentials", withdrawal_credentials.toHexString());
+          break;
+
+        case "proof_of_possession":
+          this.outputFieldMap.put("proof_of_possession", proof_of_possession.toHexString());
+          break;
+
+        case "amount":
+          this.outputFieldMap.put("amount", amount);
+          break;
+
+        case "merkel_tree_index":
+          this.outputFieldMap.put("merkel_tree_index", merkel_tree_index.toHexString());
+          break;
+      }
+    }
+  }
+
   @Override
   public String toJSON() {
     Gson gson = new GsonBuilder().create();
     GsonBuilder gsonBuilder = new GsonBuilder();
-
-    JsonObject deposit = new JsonObject();
-    deposit.addProperty("eventType", "Deposit");
-    deposit.addProperty("pubkey", pubkey.getPublicKey().toBytesCompressed().toHexString());
-    deposit.addProperty("withdrawal_crednetials", withdrawal_credentials.toHexString());
-    deposit.addProperty("proof_of_possession", proof_of_possession.toHexString());
-    deposit.addProperty("amount", amount);
-    deposit.addProperty("merkel_tree_index", merkel_tree_index.toHexString());
-
+    String jsonString = gson.toJson(this.outputFieldMap);
+    JsonObject deposit = gson.fromJson(jsonString, JsonObject.class);
     Gson customGson = gsonBuilder.setPrettyPrinting().create();
     return customGson.toJson(deposit);
   }
 
   @Override
   public String toCSV() {
-    return null;
+    String csvOutputString = "";
+    for (Object obj : this.outputFieldMap.values()) {
+      csvOutputString += "'" + obj.toString() + "',";
+    }
+    return csvOutputString.substring(0, csvOutputString.length() - 1);
   }
```

Comment:
Need a check to ensure that csvOutputString.length() > 0

Reply:
My Mistake. Fixed it.

Reply:
👍